sklearn/tests/test_metadata_routing.py

```diff
@@ -62,13 +62,6 @@
 my_other_weights = rng.rand(N)
 
 
-@pytest.fixture(autouse=True)
```

Comment:
@ogrisel was there no other way in the files where this needs to be always active? I don't like that every single test has it here.

Reply:
I personally lean slightly towards being explicit even if that means more repeated code but yeah I do get your point.

I guess if you really wanted to, you could do something like below. I am not sure I would recommend it though, it feels slightly too magical. One caveat is that the `for` loop needs to be at the very bottom of the file otherwise it will not decorate all test functions.

```
def decorator(f):
    def wrapped(*args, **kwargs):
        print('decorator')
        return f(*args, **kwargs)
    return wrapped

def test_1():
    pass

def test_2():
    pass

my_locals = [(name, obj) for name, obj in locals().items() if name.startswith('test_') and callable(obj)]

for name, obj in my_locals:
    locals()[name] = decorator(obj)
```

Reply:
Or we simply use a thread safe version of a pytest fixture?

Reply:
I wished "simply" applied here but unfortunately it's not the case :wink: 

One of the issue is that the fixture runs in the main thread and the test function (through the pytest plugin like pytest-run-parallel) runs inside a different thread. So if you change scikit-learn config settings in the fixture your changing the config in the main thread and that does not affect the config in the test function.

See https://github.com/scikit-learn/scikit-learn/issues/30007#issuecomment-2397202774 for more details.